### PR TITLE
child get quests and details

### PR DESCRIPTION
**Pull request by isabelleleeson**
- get quests for child (same as parent but don't show the owner cause it's always the same child)
- get quest details for child's quest (child can't edit them)
image
image

### CODE REVIEW

File: android/app/src/main/java/com/chorestory/app/ChildQuestDetailsActivity.java [hunk 1]
```diff
@@ -48,60 +65,94 @@ protected void onCreate(Bundle savedInstanceState) {
...
 
         // Recurrence
-        recurrenceTextView.setText("Repeat once a week");
+        recurrenceTextView.setText(quest.getRecurrenceText());
 
-        // TODO: fetch actual value
-        questDescriptionTextView.setText("This is the description of the quest");
+        // Description
+        questDescriptionTextView.setText(quest.getDescription());
+
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/app/ChildQuestDetailsActivity.java]: remove new line

File: android/app/src/main/java/com/chorestory/app/ChildQuestDetailsActivity.java [hunk 2]
```diff
@@ -48,60 +65,94 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                    public void onResponse(Call<GetQuestResponse> call, Response<GetQuestResponse> response) {
+                        if (response.isSuccessful() && response.body() != null && response.body().hasResponse()) {
+                            quest = new QuestRecyclerViewItem(response.body().getQuest());
+                            populateFields();
+                        }
+                    }
+
+                    @Override
+                    public void onFailure(Call<GetQuestResponse> call, Throwable t) {
+                        // TODO: delete the token we have stored and redirect the user to the login page?
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/app/ChildQuestDetailsActivity.java]: call `deleteTokenNavigateMain` once Cristian's PR is merged

File: android/app/src/main/java/com/chorestory/app/ChildQuestDetailsActivity.java [hunk 3]
```diff
@@ -48,60 +65,94 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                    }
+                });
+            } else {
+                // TODO: redirect to login page
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/app/ChildQuestDetailsActivity.java]: call `deleteTokenNavigateMain` once Cristian's PR is merged

File: android/app/src/main/java/com/chorestory/app/ChildQuestDetailsActivity.java [hunk 4]
```diff
@@ -48,60 +65,94 @@ protected void onCreate(Bundle savedInstanceState) {
...
+            }
+        }
     }
 
     private String getDateText() {
-        return mDay + "-" + mMonth + "-" + mYear;
+        return mDay + "-" + (mMonth+1) + "-" + mYear;
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/app/ChildQuestDetailsActivity.java]: `reformat code` on all files

File: android/app/src/main/java/com/chorestory/fragment/ChildQuestsFragment.java [hunk 5]
```diff
@@ -63,4 +73,44 @@ private void setupViewPager() {
...
+                        }
+                    }
+
+                    @Override
+                    public void onFailure(Call<GetQuestsResponse> call, Throwable t) {
+                        Toaster.showToast(getContext(), "Internal error occurred.");
+                    }
+                });
+            } else {
+                // TODO: redirect to login page
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/fragment/ChildQuestsFragment.java]: call `deleteTokenNavigateMain` once Cristian's PR is merged

File: android/app/src/main/java/com/chorestory/fragment/ChildQuestsFragment.java [hunk 6]
```diff
@@ -63,4 +73,44 @@ private void setupViewPager() {
...
+                                questParcelables.add(new QuestParcelable(quests.get(i)));
+                            }
+                            setupViewPager();
+                            tabLayout.setupWithViewPager(viewPager);
+                        }
+                    }
+
+                    @Override
+                    public void onFailure(Call<GetQuestsResponse> call, Throwable t) {
+                        Toaster.showToast(getContext(), "Internal error occurred.");
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/fragment/ChildQuestsFragment.java]: call `deleteTokenNavigateMain` once Cristian's PR is merged

File: android/app/src/main/java/com/chorestory/app/ChildQuestDetailsActivity.java (hunk 2 above)
Comment (isabelleleeson) [reply android/app/src/main/java/com/chorestory/app/ChildQuestDetailsActivity.java]: this method is only for fragments?

**Pull request merged**